apps/dashboard/web/src/apps/dashboard/views/marketplace/PageMarketplace.vue

```diff
@@ -183,20 +185,72 @@ export default {
                 })
             })
         },
-        nameToKvObj(names) {
-            return names.map(x => {
-                let category = x.split("/")[0]
-                let subCategory = x.split("/")[1]
+        setSearchText(searchText){
+            this.searchText = searchText
+            if(this.searchText && this.searchText != null){
+                this.onSearch()
+            }
+            this.$store.commit('marketplace/searchTestResults', { 'searchText': this.searchText })
+        },
+        async onSearch() {
+            this.searchItems = []
+            this.searchUserItems = []
 
-                let categoryShortName = this.businessCategoryShortNamesMap[category]
-                return {
-                    text: categoryShortName+"/"+subCategory,
-                    value: subCategory.toLowerCase().replaceAll(" ", "_")
-                }
+            await api.searchTestResults(this.searchText).then((resp) =>{
+                this.businessCategories = resp.categories
```

Comment:
@Ark2307 - can't understand this code. Can you please make it simpler? Also, I see some duplicated code snippets here - please put them in a new func.